From f8686d56151d2f76b50a49f41b4c58ae5ee27518 Mon Sep 17 00:00:00 2001
From: WeiguangTWK <weiguangtwk@outlook.com>
Date: Thu, 2 Jul 2026 15:04:59 +0800
Subject: [PATCH] [FOR XIAOMI ONLY] Neko LineageOS: Add mi_thermald control for
 Settings

THIS FEATURE IS STILL WORK IN PROGRESS

This feature allows user to temporaryarily replace thermal profile & restart mi_thermald to achieve faster charge

This is the settings Part

Change-Id: I1bc343ad59c28362470d116a61fe80366a929ded
---
 res/values-zh-rCN/strings.xml                                   | 2 ++
 res/values-zh-rTW/strings.xml                                   | 2 ++
 res/values/strings.xml                                          | 2 ++
 .../settings/fuelgauge/ChargingSpeedPreferenceController.java   | 1 +
 4 files changed, 7 insertions(+)

diff --git a/res/values-zh-rCN/strings.xml b/res/values-zh-rCN/strings.xml
index a64bf77..0550840 100644
--- a/res/values-zh-rCN/strings.xml
+++ b/res/values-zh-rCN/strings.xml
@@ -2886,6 +2886,8 @@
     <string name="printer_info_desc" msgid="1206872325746154206">"此打印机的详细信息"</string>
     <string name="power_usage_summary_title" msgid="4198312848584882113">"电池"</string>
     <string name="power_usage_summary" msgid="6857382582534984531">"耗电情况"</string>
+    <string name="charging_limit_relaxation_title">紧急限制放宽</string>
+    <string name="charging_limit_relaxation_summary">临时放宽充电限制，仅对当前充电会话生效。</string>
     <string name="power_usage_level_and_status" msgid="821521456989429593">"<xliff:g id="LEVEL">%1$s</xliff:g> - <xliff:g id="STATUS">%2$s</xliff:g>"</string>
     <string name="power_discharge_remaining" msgid="6997529817917076536">"还可用：<xliff:g id="REMAIN">%1$s</xliff:g>"</string>
     <string name="power_charge_remaining" msgid="7046064326172265116">"充电剩余时间：<xliff:g id="UNTIL_CHARGED">%1$s</xliff:g>"</string>
diff --git a/res/values-zh-rTW/strings.xml b/res/values-zh-rTW/strings.xml
index 1b4eb7d..6677075 100644
--- a/res/values-zh-rTW/strings.xml
+++ b/res/values-zh-rTW/strings.xml
@@ -2886,6 +2886,8 @@
     <string name="printer_info_desc" msgid="1206872325746154206">"查看這台印表機的詳細資訊"</string>
     <string name="power_usage_summary_title" msgid="4198312848584882113">"電池"</string>
     <string name="power_usage_summary" msgid="6857382582534984531">"查看正在消耗電力的功能"</string>
+    <string name="charging_limit_relaxation_title">緊急限制放寬</string>
+    <string name="charging_limit_relaxation_summary">暫時放寬充電限制，僅對目前充電工作階段生效。</string>
     <string name="power_usage_level_and_status" msgid="821521456989429593">"<xliff:g id="LEVEL">%1$s</xliff:g> - <xliff:g id="STATUS">%2$s</xliff:g>"</string>
     <string name="power_discharge_remaining" msgid="6997529817917076536">"還剩 <xliff:g id="REMAIN">%1$s</xliff:g>"</string>
     <string name="power_charge_remaining" msgid="7046064326172265116">"<xliff:g id="UNTIL_CHARGED">%1$s</xliff:g>後充電完畢"</string>
diff --git a/res/values/strings.xml b/res/values/strings.xml
index 61408fa..6195d42 100644
--- a/res/values/strings.xml
+++ b/res/values/strings.xml
@@ -6712,6 +6712,8 @@
     <string name="power_usage_summary_title">Battery</string>
     <!-- Activity title summary for App Fuel Gauge summary -->
     <string name="power_usage_summary">What has been using the battery</string>
+    <string name="charging_limit_relaxation_title">Emergency limit relaxation</string>
+    <string name="charging_limit_relaxation_summary">Temporarily relaxes charging limits for the current charge session only.</string>
     <!-- Display the battery level and status [CHAR_LIMIT=60] -->
     <string name="power_usage_level_and_status"><xliff:g id="level">%1$s</xliff:g>
         - <xliff:g id="status">%2$s</xliff:g></string>
diff --git a/src/com/android/settings/fuelgauge/ChargingSpeedPreferenceController.java b/src/com/android/settings/fuelgauge/ChargingSpeedPreferenceController.java
index a120ca6..b429fe1 100644
--- a/src/com/android/settings/fuelgauge/ChargingSpeedPreferenceController.java
+++ b/src/com/android/settings/fuelgauge/ChargingSpeedPreferenceController.java
@@ -83,6 +83,7 @@ public class ChargingSpeedPreferenceController extends BasePreferenceController
     @Override
     public int getAvailabilityStatus() {
         return mHealthInterface != null && mHealthInterface.isFastChargeSupported()
+                && mHealthInterface.getSupportedFastChargeModes().length > 1
                 ? AVAILABLE : UNSUPPORTED_ON_DEVICE;
     }
 
-- 
2.43.0

